plugins/kubernetes-backend/api-report.md

```diff
@@ -22,51 +22,60 @@ import type { RequestHandler } from 'express';
 import { TokenCredential } from '@azure/identity';
 
 // @public (undocumented)
-export class AksKubernetesAuthTranslator {
+export class AksStrategy implements AuthenticationStrategy {
   // (undocumented)
-  decorateClusterDetailsWithAuth(
-    clusterDetails: ClusterDetails,
-    auth: KubernetesRequestAuth,
-  ): Promise<ClusterDetails>;
+  getCredential(
+    _: ClusterDetails,
+    requestAuth: KubernetesRequestAuth,
+  ): Promise<KubernetesCredential>;
+  // (undocumented)
+  validate(_: AuthMetadata): void;
 }
 
 // @public (undocumented)
-export interface AWSClusterDetails extends ClusterDetails {
+export class AnonymousStrategy implements AuthenticationStrategy {
   // (undocumented)
-  assumeRole?: string;
+  getCredential(): Promise<KubernetesCredential>;
   // (undocumented)
-  externalId?: string;
+  validate(): void;
 }
 
 // @public (undocumented)
-export class AwsIamKubernetesAuthTranslator
-  implements KubernetesAuthTranslator
-{
-  constructor(opts: { config: Config });
+export interface AuthenticationStrategy {
+  // (undocumented)
+  getCredential(
+    clusterDetails: ClusterDetails,
+    authConfig: KubernetesRequestAuth,
+  ): Promise<KubernetesCredential>;
   // (undocumented)
-  decorateClusterDetailsWithAuth(
-    clusterDetails: AWSClusterDetails,
-  ): Promise<AWSClusterDetails>;
+  validate(authMetadata: AuthMetadata): void;
```

Comment:
Perhaps we should return a list of error objects instead if using exceptions?

Reply:
201a86e6a2a60adbac1c85a25742739db7ee7dbb